### PR TITLE
Switch to (almost) vanilla libp2p

### PR DESCRIPTION
**Pull request by folex**
Waiting on two PRs before can use libp2p from crates.io:
- https://github.com/libp2p/rust-libp2p/pull/2437
- https://github.com/libp2p/rust-libp2p/pull/2436

Before these PRs, will continue to use libp2p from fluencelabs/rust-libp2p fork on `libp2p_core_disable_default_features` branch. It differs from upstream by just a few lines.

### CODE REVIEW

File: aquamarine/Cargo.toml [hunk 1]
```diff
@@ -17,7 +17,8 @@ fs-utils = { path = "../crates/fs-utils" }
 particle-execution = { path = "../particle-execution" }
 
 avm-server = "0.13.0"
-libp2p = { version = "0.36.2", package = "fluence-fork-libp2p", features = ["tcp-tokio"] }
+libp2p = "0.42"
+
```
Comment (justprosh) [inline aquamarine/Cargo.toml:21]:
```suggestion
```

File: connection-pool/Cargo.toml [hunk 2]
```diff
@@ -6,10 +6,10 @@ edition = "2018"
 
 [dependencies]
 particle-protocol = { path = "../particle-protocol"}
-trust-graph = "0.2.7"
 fluence-libp2p = { path = "../crates/libp2p" }
 
-libp2p = { version = "0.36.2", package = "fluence-fork-libp2p", features = ["tcp-tokio"] }
+libp2p = "0.42"
+
```
Comment (justprosh) [inline connection-pool/Cargo.toml:12]:
```suggestion
```

File: crates/kademlia/Cargo.toml [hunk 3]
```diff
@@ -7,14 +7,14 @@ edition = "2018"
...
-trust-graph = "0.2.7"
 control-macro = { path = "../control-macro" }
 fluence-libp2p = { path = "../libp2p" }
 server-config = { path = "../server-config" }
 
-libp2p = { version = "0.36.2", package = "fluence-fork-libp2p", features = ["tcp-tokio"] }
+libp2p = "0.42"
 
-multihash = "0.13.2"
+
```
Comment (justprosh) [inline crates/kademlia/Cargo.toml:16]:
```suggestion
```

File: crates/kademlia/src/behaviour.rs [hunk 4]
```diff
@@ -120,25 +122,15 @@ pub struct Kademlia {
...
-            store,
-            config.as_libp2p(),
-            trust_graph,
-        );
+        let mut kademlia = kad::Kademlia::with_config(config.peer_id, store, config.as_libp2p());
 
-        if let Some(registry) = registry {
-            kademlia.enable_metrics(registry);
-        }
+        // if let Some(registry) = registry {
```
Comment (justprosh) [inline crates/kademlia/src/behaviour.rs:131]: Why is this commented?

File: crates/local-vm/Cargo.toml [hunk 5]
```diff
@@ -18,7 +18,8 @@ air-interpreter-fs = { path = "../air-interpreter-fs" }
 air-interpreter-wasm = "=0.19.8"
 avm-server = "0.13.0"
 
-libp2p = { version = "0.36.1", package = "fluence-fork-libp2p", features = ["tcp-tokio"] }
+libp2p = "0.42"
+
```
Comment (justprosh) [inline crates/local-vm/Cargo.toml:22]:
```suggestion
```

**Review by justprosh — APPROVED** [review]

File: crates/kademlia/src/behaviour.rs (hunk 4 above)
Comment (folex) [reply crates/kademlia/src/behaviour.rs:132]: Because `kademlia_enable.metrics` was only implemented in our fork of libp2p. It's addressed in the next PR.

**Comment by folex** [comment]
Superseded by #1245

**Pull request closed without merging**